### PR TITLE
Add performance tests for verifying blocks full of transparent transactions

### PR DESCRIPTION
**Pull request by str4d**
Related to #971 (but not what that issue requests).

### CODE REVIEW

**Comment by str4d** [comment]
I have added two benchmarks - one with a block full of chained transactions each signed by the coinbase address, and the other with a block full of chained anyone-can-spend transactions. Surprisingly (to me), the anyone-can-spend transactions take longer to validate.

**Comment by zookozcash** [comment]
Way to go, str4d!

File: src/zcbenchmarks.cpp [hunk 1]
```diff
@@ -126,3 +132,107 @@ double benchmark_verify_equihash()
...
+    // 3) Call TestBlockValidity again under timing
+    CBlockIndex* pindexPrev = chainActive.Tip();
+    CValidationState state;
+    timer_start();
+    assert(TestBlockValidity(state, *pblock, pindexPrev, false, false));
+    double ret = timer_stop();
+    delete pblocktemplate;
+    mempool.clear();
+    return ret;
+}
```
Comment (daira) [inline src/zcbenchmarks.cpp]: Refactor this into a single function that takes an ``anyonecanspend`` flag.

**Comment by str4d** [comment]
#1046 contains the test that @nathan-at-least actually wanted. Leaving this here in case it is still useful, in which case I can refactor after #1046 (the more important test) is merged.

**Comment by zookozcash** [comment]
What does this PR do that pull request #1046 doesn't do? AFAICT this just benchmarks the time to verify a block full of transparent transactions.

**Comment by ebfull** [comment]
I plan to close this PR because it's out of date and I suspect we don't need it anymore. Any objections?

**Comment by zookozcash** [comment]
I object! I don't see a measurement on https://speed.z.cash/timeline/ that shows me the time that it takes to verify a block full of transparent transactions.

**Comment by ebfull** [comment]
Did you miss https://speed.z.cash/timeline/?exe=1&base=1%2B9&ben=time+validatelargetx&env=1&revs=50&equid=off&quarts=on&extr=on ?

**Comment by zookozcash** [comment]
Yes! Thanks. Okay, I guess we can close this ticket as "fixed".

**Pull request closed without merging**